pkg/importer/util.go

```diff
@@ -52,3 +54,28 @@ func StreamDataToFile(dataReader io.Reader, filePath string) error {
 	}
 	return nil
 }
+
+// StreamDataToBlockDevice provides a function to stream the specified io.Reader to the specified block device
+func StreamDataToBlockDevice(dataReader io.Reader, filePath string) error {
+	var fd int
+	var err error
+
+	fd, err = syscall.Open(filePath, os.O_CREATE|os.O_WRONLY, 0777)
+	defer syscall.Close(fd)
+	if err != nil {
+		return errors.Wrapf(err, "could not open device file %q", filePath)
+	}
+
+	buf := new(bytes.Buffer)
+	glog.V(1).Infof("reading from dataReader....\n")
+	buf.ReadFrom(dataReader)
+	glog.V(1).Infof("writing to block device....\n")
+	if _, err = syscall.Write(fd, buf.Bytes()); err != nil {
+		syscall.Unlink(filePath)
+		return errors.Wrapf(err, "unable to write to device file")
+	}
+	if err := syscall.Fsync(fd); err != nil {
```

Comment:
I suspect it would be more efficient to open the file for direct I/O which will bypass the buffer cache and makes fsync unnecessary.  But using directio can be a little tricky as buffers have to be alligned.  May be worth benchmarking.

Reply:
so the impression I got from directI/O was bad:
https://stackoverflow.com/a/33101119/8393409
What do you think? I might be wrong...

Reply:
Linus hates c++, so should we not use programs written in c++?  He's also complaining about the "interface" to directio which requires the aligned buffers I mention earlier.  I think it's pretty dumb that the buffers have to be aligned, but it is what it is.  And if you want to skip the os buffer cache (should prob benchmark to decide if we should) it's the easiest way to go.

Reply:
changed the code to limit size of buffer as io.copy implemented